src/chain/chain.rs

```diff
@@ -433,11 +436,7 @@ impl Chain {
     fn poll_relocation(&mut self) -> Option<RelocateDetails> {
         // Delay relocation until all backlogged churn events have been handled and no
         // additional churn is in progress. Only allow one relocation at a time.
-        if self.relocation_in_progress
-            || self.churn_in_progress
-            || !self.state.churn_event_backlog.is_empty()
-            || !self.state.handled_genesis_event
-        {
+        if !self.can_poll_churn() || !self.state.churn_event_backlog.is_empty() {
```

Comment:
just to confirm that the `can_poll_churn` function will check additional flag `split_in_progress` from state.
Is this ok?

Reply:
Yes, really split_in_progress will only be true if churn_in_progress is true also, so does not make much difference.
This set of flags can be cleaned up / improved later.